src/test/java/com/statful/client/StatfulHttpClientIntegrationTest.java

```diff
@@ -0,0 +1,165 @@
+package com.statful.client;
+
+import com.google.common.collect.Lists;
+import com.statful.utils.Pair;
+import io.vertx.core.Vertx;
+import io.vertx.core.VertxOptions;
+import io.vertx.core.http.HttpServer;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import io.vertx.ext.unit.Async;
+import io.vertx.ext.unit.TestContext;
+import io.vertx.ext.unit.junit.VertxUnitRunner;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static java.util.Objects.nonNull;
+
+@RunWith(VertxUnitRunner.class)
+public class StatfulHttpClientIntegrationTest extends IntegrationTestCase {
```

Comment:
They are, the problem with that is the receiver has different logic, or do you mean having them on the same class?